module_analysis/models/ir_module_module.py

```diff
@@ -0,0 +1,178 @@
+# Copyright (C) 2019-Today: GRAP (<http://www.grap.coop/>)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+import os
+from pathlib import Path
+
+from pygount import SourceAnalysis
+
+from odoo import api, fields, models
+from odoo.modules.module import get_module_path
+from odoo.tools.safe_eval import safe_eval
+
+_logger = logging.getLogger(__name__)
+
+
+class IrModuleModule(models.Model):
+    _inherit = "ir.module.module"
+
+    author_ids = fields.Many2many(
+        string="Authors",
+        comodel_name="ir.module.author",
+        readonly=True,
+        relation="ir_module_module_author_rel",
+    )
+
+    module_type_id = fields.Many2one(
+        string="Module Type", comodel_name="ir.module.type", readonly=True
+    )
+
+    python_code_qty = fields.Integer(string="Python Code Quantity", readonly=True)
+
+    xml_code_qty = fields.Integer(string="XML Code Quantity", readonly=True)
+
+    js_code_qty = fields.Integer(string="JS Code Quantity", readonly=True)
+
+    css_code_qty = fields.Integer(string="CSS Code Quantity", readonly=True)
+
+    # Overloadable Section
+    @api.model
+    def _get_analyse_settings(self):
+        """Return a dictionnary of data analysed
+        Overload this function if you want to analyse other data
+        {
+            'extension': {
+                'data_to_analyse': 'field_name',
+            },
+        }, [...]
+        extension: extension of the file, with the '.'
+        data_to_analyse : possible value : code, documentation, empty, string
+        field_name: Odoo field name to store the analysis
+        """
+        return {
+            ".py": {"code": "python_code_qty"},
+            ".xml": {"code": "xml_code_qty"},
+            ".js": {"code": "js_code_qty"},
+            ".css": {"code": "css_code_qty"},
```

Comment:
Doest it cover scss too?

Reply:
it seems there is about a 50/50 split between CSS and SCSS files in the OCA, so it seems reasonable to add

Reply:
done here https://github.com/OCA/server-tools/pull/2085/commits/ecbfdcbcab9a2182bed5d57c52330bae32e71d85